### PR TITLE
Misc updates

### LINKED ISSUE #314: Need to transport CL and BR
Chemical species CL and BR need to be transported in all configurations.

We have not been transporting CL and BR in the WACCM middle atmosphere configurations, including D-region chemistry configuration, (waccm_ma and waccm_mad). The assumption that these species do not need to be transported breaks down in the upper atmosphere (above 80 km or so) where species are not well mixed.  This has lead to unrealistic concentrations of chlorine and bromine simulations that use WACCM middle atmosphere chemistry.

The chemistry mechanisms that need to change are:
-  waccm_ma
-  waccm_ma_mam4
-  waccm_mad
-  waccm_mad_mam4

### LINKED ISSUE #317: error in mo_srf_emissions using ESMF library with leap year
When running the aux_cam testlist on cheyenne - the following test:
SMS_D_Ln9_Vnuopc.f09_f09_mg17.FCts2nudged.cheyenne_intel.cam-outfrq9s_leapday
Resulted in the following traceback in the cesm.log file
```
40:cesm.exe           000000000B757464  shr_abort_mod_mp_         114  shr_abort_mod.F90
40:cesm.exe           000000000B7572C2  shr_abort_mod_mp_          61  shr_abort_mod.F90
40:cesm.exe           0000000002A9EB37  time_manager_mp_c        1242  time_manager.F90
40:cesm.exe           0000000002A93AAD  time_manager_mp_s         271  time_manager.F90
40:cesm.exe           0000000002AEBCB9  tracer_data_mp_re        1200  tracer_data.F90
40:cesm.exe           0000000002AE121E  tracer_data_mp_ad         704  tracer_data.F90
40:cesm.exe           0000000006099E17  mo_srf_emissions_         309  mo_srf_emissions.F90
40:cesm.exe           000000000320C071  chemistry_mp_chem        1102  chemistry.F90
40:cesm.exe           000000000116DFD1  physpkg_mp_phys_t        2715  physpkg.F90
40:cesm.exe           0000000001139B93  physpkg_mp_phys_r        1036  physpkg.F90
40:cesm.exe           00000000007D8D8A  cam_comp_mp_cam_r         258  cam_comp.F90
40:cesm.exe           0000000000786E10  atm_comp_nuopc_mp         845  atm_comp_nuopc.F90
```

The atm log file contained the following output before the crash:
 ```
emiss_shipping
 READ_NEXT_TRCDATA emiss_bb
 READ_NEXT_TRCDATA emiss_res_tran
 set_time_float_from_date: error return from ESMF_TimeSet for set_time_float_from_date
 ERROR: CHKRC
```
And finally, the PET log file contained the following error message - which points to the source of the problem.
 PET0222 ESMCI_Calendar.C:1059 ESMCI::Calendar::convertToTime() Input argument out of range  - ; Gregorian: for Febru\ary 2010, dd=29 > 28 days in the month

It seems that the mct esmf_wrf_timemgr is not picking up these type of failures (the same issue arose in a CTSM test) and using the ESMF library for time management triggers the expected error.

### LINKED ISSUE #321: photolysis maximum solar zenith
Maximum solar zenith angle used in photochemistry should be the same for cam-chem and waccm.

### LINKED ISSUE #446: Failing WACCMX test
As of tag cam6_3_031, the following WACCMX aux_cam test fails to run:
SMS_D_Ln9_Vmct.f19_f19_mg17.FXHIST.cheyenne_intel.cam-outfrq9s_amie 

The suspicion is that the updated externals in this tag has caused the job to require more memory.

### PR DESCRIPTION
**Pull request by fvitt**
This is combined PR addresses:
 - A leap day issue when ESMF lib is used in CMEP coupled compsets
 - Increase maximum solar zenith angle used in photolysis in CAM-Chem
 - Transport BR and CL in WACCM middle atmosphere chemistry
 - Increase tasks count in default PE layouts for WACCMX on cheyenne

fixes #314
fixes #317 
fixes #321 
fixes #446

### CODE REVIEW

File: src/chemistry/mozart/mo_photo.F90 [hunk 1]
```diff
@@ -173,25 +174,22 @@ subroutine photo_inti( xs_coef_file, xs_short_file, xs_long_file, rsf_file, &
...
-       max_zen_angle = 116._r8
-    else if ( ptop_ref < 10._r8 ) then
-       max_zen_angle = 97.01_r8 ! degrees
-    else
-       max_zen_angle = 88.85_r8 ! degrees
-    endif
+    ! maximum solar zenith angle for which photo-chemical rates are computed
+    max_zen_angle = maxzen
+    if (.not. max_zen_angle>0._r8) then
+       write(*,*) 'photo_inti: max_zen_angle = ',max_zen_angle
```
Comment (nusbaume) [inline src/chemistry/mozart/mo_photo.F90:180]: Should this be written to `iulog`?

File: src/chemistry/pp_waccm_mad/chem_mech.in [hunk 2]
```diff
@@ -147,6 +145,7 @@
  e,
  H3Op_OH,
  HCO3m,
+ HO2,
```
Comment (gold2718) [inline src/chemistry/pp_waccm_mad/chem_mech.in:148]: Is this something unrelated to Br or Cl? If so, can you document it in the ChangeLog file?
I think this applies to some of the other chemistry packages as well and to other items such as CO2 below.

File: doc/ChangeLog [hunk 3]
```diff
@@ -1,4 +1,178 @@
 ===============================================================
+
+Tag name:
+Originator(s): fvitt
+Date:
+One-line Summary: Transport CL BR; max solar zenth angle; waccmx pelayout; fix leap day issue
+Github PR URL:
```
Comment (gold2718) [inline doc/ChangeLog:7]: This can be filled in.

File: doc/ChangeLog [hunk 4]
```diff
@@ -1,4 +1,178 @@
...
+
+Purpose of changes (include the issue number and title text for each relevant GitHub issue):
+
+ Address github issues:
+   . error in mo_srf_emissions using ESMF library with leap year #317
+   . Need to transport CL and BR #314
+   . photolysis maximum solar zenith #321
+   . Failing WACCMX test #446
+
+Describe any changes made to build system:
```
Comment (gold2718) [inline doc/ChangeLog:17]: Add NA here.

File: doc/ChangeLog [hunk 5]
```diff
@@ -1,4 +1,178 @@
...
+
+Describe any changes made to the namelist:
+ New photo_max_zen namelist option:
+ Maximum zenith angle (degrees) used for photolysis.
+
+List any changes to the defaults for the boundary datasets:
+
+Describe any substantial timing or memory changes:
```
Comment (gold2718) [inline doc/ChangeLog:25]: Add NA (or none) when no changes were made.

**Review by gold2718 — APPROVED** [review]
Looks good in general. I have a few questions or change requests for the ChangeLog.

File: src/chemistry/pp_waccm_ma_mam4/chem_mech.in [hunk 6]
```diff
@@ -551,6 +550,8 @@
 *********************************
 *** Sulfur
 *********************************
+[DMS_NO3]              DMS + NO3  -> SO2 + HNO3                                 ; 1.9e-13, 520 
+[DMS_OHa]              DMS + OH  -> SO2                                         ; 9.6e-12, -234
```
[the remark this answers is not in the document]
Comment (fvitt) [reply src/chemistry/pp_waccm_ma_mam4/chem_mech.in:554]: The order of species and reactions has changed in some of the mechanism.  I will document this in the ChangeLog file.

File: src/chemistry/pp_waccm_mad/chem_mech.in (hunk 2 above)
Comment (fvitt) [reply src/chemistry/pp_waccm_mad/chem_mech.in:148]: Right. This is unrelated to Br or Cl transport.  These extra changes are done to be consistent across all waccm mechanisms.  Will document these in ChangeLog.

File: doc/ChangeLog (hunk 4 above)
Comment (fvitt) [reply doc/ChangeLog:17]: done

File: doc/ChangeLog (hunk 5 above)
Comment (fvitt) [reply doc/ChangeLog:25]: done

File: doc/ChangeLog (hunk 3 above)
Comment (fvitt) [reply doc/ChangeLog:7]: done

File: doc/ChangeLog [hunk 7]
```diff
@@ -1,4 +1,178 @@
...
+A       bld/namelist_files/use_cases/aquaplanet_waccm_2000.xml
+ - for aqua-planet waccm
+
+List all existing files that have been modified, and describe the changes:
+
+M       .gitignore
+ - ignore libraries/mct, libraries/parallelio, and share externals
+
+M       bld/build-namelist
+ - add logic to set max solar zeneth angle for photolysis
```
[the remark this answers is not in the document]
Comment (fvitt) [reply doc/ChangeLog:57]: done

File: src/chemistry/mozart/mo_photo.F90 (hunk 1 above)
Comment (fvitt) [reply src/chemistry/mozart/mo_photo.F90]: Done

**Pull request merged**